### PR TITLE
[libSyntax] Inline commonly called methods in RawSyntax and AbsoluteRawSyntax

### PR DESCRIPTION
**Pull request by ahoppen**
These methods are small and setting up the stack frame etc. takes up the majority (or at least a significant amount) of their execution time. So let's give the compiler a chance to inline them.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: include/swift/Syntax/RawSyntax.h [hunk 1]
```diff
@@ -120,6 +120,27 @@ using llvm::StringRef;
 #define syntax_assert_token_is(Tok, Kind, Text)
 #endif
 
+namespace {
+/// If the \p Str is not allocated in \p Arena, copy it to \p Arena and adjust
+/// \p Str to point to the string's copy in \p Arena.
+void copyToArenaIfNecessary(
+    StringRef &Str, const swift::RC<swift::syntax::SyntaxArena> &Arena) {
```
Comment (rintaro) [inline include/swift/Syntax/RawSyntax.h:127]: Could this be a `SyntaxArena` method? I'm just uncomfortable with having an anonymous namespace in a header.

**Review by rintaro — COMMENTED** [review]
I'm concerned about the code size. Especially, `RawSyntax::make()` (via `makeAndCalcLength()`) are called from many places including `SyntaxFactory` and `SyntaxBuilder`. Do you know the actual code side change?

File: include/swift/Syntax/RawSyntax.h (hunk 1 above)
Comment (ahoppen) [reply include/swift/Syntax/RawSyntax.h:127]: Good idea. Then we could also use this to copy the entire source code over here: https://github.com/apple/swift/blob/main/lib/SyntaxParse/SyntaxTreeCreator.cpp#L42

**Comment by ahoppen** [comment]
> I'm concerned about the code size

I just checked and defining the `RawSyntax` constructors and `make` function in the header does not seem to have a noticeable on file size (in release builds, didn’t try other variants)
```
Size of swift-frontend with constructors and make functions defined in 
RawSyntax.h:   94959944 bytes
RawSyntax.cpp: 94960688 bytes
```

AFAIU defining a function in a header (or alternatively making it `inline`) allows the compiler to inline the function, but it’s not required to do so if it doesn’t think it’s beneficial, which is presumably what happens in `SyntaxFactory`.

From https://en.cppreference.com/w/cpp/language/inline
> Since this meaning of the keyword inline is non-binding, compilers are free to use inline substitution for any function that's not marked inline, and are free to generate function calls to any function marked inline. Those optimization choices do not change the rules regarding multiple definitions and shared statics listed above.

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test macOS

**Review by rintaro — APPROVED** [review]
Confirmed the binary size doesn't change much :)

**Pull request merged**